src/redux.js

```diff
@@ -633,6 +633,45 @@ export function getSelectedColumns(state) {
     });
 }
 
+function isColumnCategorical(state, column) {
+  return (state.columnsByDataType[column] === ColumnTypes.CATEGORICAL);
+}
+
+function isValidCategoricalData(state, column) {
+  return !hasTooManyUniqueOptions(state, column);
+}
+
+function isColumnNumerical(state, column) {
+  return (state.columnsByDataType[column] === ColumnTypes.NUMERICAL);
+}
+
+function isValidNumericalData(state, column) {
+  return !isNaN(getMaximumValue(state, column));
+}
+
+function isColumnDataValid(state, column) {
+  return (
+    isColumnCategorical(state, column) && isValidCategoricalData(state,column)
+  ) ||
+  (isColumnNumerical(state, column) && isValidNumericalData(state,column));
+}
```

Comment:
My intent is that this structure will allow us to layer in a check for empty cells, or anything else we find mucks up the data more easily, but definitely open to feedback!

Reply:
This seems fine for now, but if it gets much longer, I like the idea of a series of separate `if` statements and then a default return value if none of them hit.